src/main/kotlin/org/rust/cargo/runconfig/RustApplicationConfiguration.kt

```diff
@@ -0,0 +1,59 @@
+package org.rust.cargo.runconfig
+
+import com.intellij.execution.Executor
+import com.intellij.execution.configurations.ModuleBasedConfiguration
+import com.intellij.execution.configurations.RunConfiguration
+import com.intellij.execution.configurations.RunConfigurationModule
+import com.intellij.execution.configurations.RunProfileState
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.options.SettingsEditor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ModuleRootManager
+import com.intellij.util.PathUtil
+import com.intellij.util.xmlb.XmlSerializer
+import org.jdom.Element
+import org.rust.cargo.project.util.getModules
+
+class RustApplicationConfiguration(project: Project,
+                                   name: String,
+                                   configurationType: RustApplicationRunConfigurationType)
+
+    : ModuleBasedConfiguration<RustRunConfigurationModule>(name,
+                                                           RustRunConfigurationModule(project),
+                                                           configurationType.configurationFactories[0]) {
+
+    var isRelease: Boolean = false
+
+    override fun getValidModules(): Collection<Module> = project.getModules()
+
+    override fun checkConfiguration() {
+        super.checkConfiguration()
+        configurationModule.checkForWarning()
+    }
+
+    override fun getConfigurationEditor(): SettingsEditor<out RunConfiguration> =
+        RustRunConfigurationEditorForm()
+
+    override fun getState(executor: Executor, environment: ExecutionEnvironment): RunProfileState  {
+        val module = configurationModule.module!!
+        val moduleManager = ModuleRootManager.getInstance(module)
+        val sdk = moduleManager.sdk!!
```

Comment:
Ah, they say just the opposite. I should return `null` if it is impossible to create a `State`. Thanks!